docs: change definition of providedIn any

santoshyadavdev opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

santoshyadav198613 commented:
@kapunahelewong @gkalpak ,
can you review, its small change.

kapunahelewong commented on packages/core/src/di/injectable.ts:
```diff
@@ -70,8 +70,8 @@ export interface Injectable {
    * applications on the page.
-   * - 'any' : Provides a unique instance in every module (including lazy modules) that injects the
-   * token.
+   * - 'any' : Provides a unique instance in every lazy loaded modules and all eager loaded module
```
```suggestion
   * - 'any' : Provides a unique instance in each lazy loaded module while all eagerly loaded modules
```

Is this the definition? That it will give a unique instance for each lazy module, but that all eagerly loaded ones will share one? (I'm still learning this.)

santoshyadav198613 replied:
Yeah, this is what i wanted to write, thanks :)

kapunahelewong reviewed (changes requested):
Thanks for this, @santoshyadav198613. Just a note on clarification.

mary-poppins commented:
You can preview 5e40996 at https://pr35292-5e40996.ngbuilds.io/.
You can preview f3f04f7 at https://pr35292-f3f04f7.ngbuilds.io/.
You can preview 3a711e1 at https://pr35292-3a711e1.ngbuilds.io/.

kapunahelewong commented on packages/core/src/di/injectable.ts:
```diff
@@ -70,8 +70,8 @@ export interface Injectable {
-   * - 'any' : Provides a unique instance in every module (including lazy modules) that injects the
-   * token.
+   * - 'any' : Provides a unique instance in each lazy loaded module while all eagerly loaded
+   * modules will share an unique instance.
```
I must have missed this before. Pardon me. Can we try this?
```suggestion
   * modules share one instance.
```

santoshyadav198613 replied:
Done.

mary-poppins commented:
You can preview 809103b at https://pr35292-809103b.ngbuilds.io/.

santoshyadav198613 commented:
Hi @kara ,
Can you review please.

santoshyadav198613 commented:
Hi @kara ,
Can you review please.

santoshyadav198613 commented:
Hi Team,
Can you review it, please?

santoshyadav198613 commented:
H @docs-team can you check and approve.

santoshyadav198613 commented:
Hi, @mhevery and @aikidave can you review, please.

IgorMinar reviewed (approved):
Lgtm, thanks! This is a helpful clarification.

IgorMinar reviewed (approved):
Reviewed-for: global-approvers

Splaktar reviewed (approved):
LGTM, thank you!

Splaktar commented:
I believe that the eng review has been completed?

It's unclear what cleanup is still needed?

AndrewKushnir commented:
@Splaktar, thanks for the comment. I've added the "cleanup" label because this PR required a rebase (merge script indicated that). I tried to rebase myself, but got the "permission denied" error.

@santoshyadav198613, could you please rebase this PR on top of the latest master and let me know in this thread, so I can try to merge it again.

Thank you.

santoshyadav198613 commented:
> @Splaktar, thanks for the comment. I've added the "cleanup" label because this PR required a rebase (merge script indicated that). I tried to rebase myself, but got the "permission denied" error.
> 
> @santoshyadav198613, could you please rebase this PR on top of the latest master and let me know in this thread, so I can try to merge it again.
> 
> Thank you.

Let me do it.

santoshyadavdev committed:
docs: change definition of providedIn any

change in the definition of providedIn:any any instance creates a singleton instance
for each lazy loaded module and one instance for eager loaded module

mary-poppins commented:
You can preview f552e99 at https://pr35292-f552e99.ngbuilds.io/.

santoshyadav198613 commented:
@AndrewKushnir done, you can merge now.
